apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -184,3 +256,95 @@ func validateImageURL(imageURL string) error {
 
 	return nil
 }
+
+// When making changes to this function for operationalstatus and errortype,
+// also make the corresponding changes in the OperationalStatus and
+// ErrorType fields in the struct definition of BareMetalHostStatus in
+// the file baremetalhost_types.go
+func checkStatusAnnotation(bmhstatus *BareMetalHostStatus) error {
+
+	operatstatus_allowed := []string{"", "OK", "discovered", "error", "delayed", "detached"}
```

Comment:
Please use defined [constants](https://github.com/metal3-io/baremetal-operator/blob/main/apis/metal3.io/v1alpha1/baremetalhost_types.go#L108-L160).

Reply:
I have used the constants although had to typecast them to string, does it look okay now ?

Reply:
Yes, but there's no need to cast everything - the point of using slices.Contains is that it can take a slice of e.g. `OperationalStatus` and search for `bmhstatus.OperationalStatus` in it.